modules/develop/pages/data-transforms.adoc

```diff
@@ -0,0 +1,253 @@
+= Data Transforms
+:description: Learn how to run WebAssembly data transforms within Redpanda brokers.
+:page-aliases: reference:rpk/rpk-wasm/rpk-wasm.adoc, reference:rpk/rpk-wasm.adoc, reference:rpk/rpk-wasm/rpk-wasm-deploy.adoc, reference:rpk/rpk-wasm/rpk-wasm-generate.adoc, reference:rpk/rpk-wasm/rpk-wasm-remove.adoc, data-management:data-transform.adoc
+:latest-data-transforms-version: 0.0.0-20230830git604fcce
+
+Redpanda provides the framework to build and deploy inline transformations on data written to Redpanda topics, delivering processed and validated data to consumers in the format they expect. Data transforms let you run common data streaming tasks, like filtering, scrubbing, and transcoding, within the Redpanda broker. For example, you may have consumers that require you to remove credit card numbers or convert JSON to Avro. Data transforms can also interact with the Redpanda Schema Registry to work with encoded data types. Redpanda does this directly inside the broker, eliminating the need to manage a separate stream processing environment or use 3rd-party tools. 
+
+image::shared:wasm1.png[Data transforms in a broker] 
+
+Data transforms use a WebAssembly (Wasm) engine inside a Redpanda broker. They run asynchronously from the source topic to the destination topic, which allows Redpanda to control the entire transform lifecycle. For example, Redpanda can stop and start transforms when partitions are moved or to free up system resources for other tasks. You can develop and manage data transforms with `rpk transform` commands.
+
+== Limitations
+
+- Transforms have no external access to disk or network resources. 
+- Only single record transforms are supported, not aggregations or joins. In other words, a Wasm function acts on a single record in an input topic.
+- Only a single output topic is supported.
+- Transforms have at-least-once delivery.
+- Because data transforms are powered by Wasm, transform functions can be authored in any language. However, libraries to author transforms currently are only available in https://go.dev/doc/install[Golang^]. 
+
+== How data transforms work
+
+Within Repdanda, there's a single Raft controller that manages cluster information, including data transforms. On every shard (core), Redpanda knows what data transforms exist in the cluster, as well as metadata about the transform, such as input and output topics and environment variables. Data transforms take an input topic and map it to an output topic, running a virtual machine (VM) on the same shard as the input partition leader. 
```

Comment:
> Within Repdanda, there's a single Raft controller that manages cluster information,

Is there anything else in the docs we can link to about the controller?

Reply:
Not really. The best doc is [here](https://docs.redpanda.com/current/get-started/architecture/#controller-partition-and-snapshots), but we do have a [WIP glossary](https://docs.google.com/document/d/17hN0HzGwNyIOEGXFqMSzHNmWZipRW6joWRjtOWUgsx4/edit) with an entry for it. We could link to it when the glossary is live (December), which would also add hover text to the term. Would it help if the diagram use Controller instead of Raft?

Reply:
No I think it's fine as is, but I think it's a nice improvement for later when we have a link to drop here.